### PR TITLE
[Backport release-24.05] percona: update packages, init 8.4

### PR DESCRIPTION
**Pull request by osnyx**
backport of #346027.

Main difference is that the `_lts` package names continue to point at the 8_0 release stream, to avoid a major release change within the 24.05 release cycle.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by osnyx** [comment]
Updated the backport according to the changes in #346027.

File: pkgs/servers/sql/percona-server/8_0.nix [hunk 1]
```diff
@@ -14,8 +49,16 @@ stdenv.mkDerivation (finalAttrs: {
...
+  nativeBuildInputs = [
+    bison
+    cmake
+    pkg-config
+    makeWrapper
+    # required for scripts/CMakeLists.txt
+    coreutils
+    gnugrep
+    procps
+  ] ++ lib.optionals (!stdenv.hostPlatform.isDarwin) [ rpcsvc-proto ];
```
Comment (Ma27) [inline pkgs/servers/sql/percona-server/8_0.nix:61]: This is not the diff from before on 24.05, but includes https://github.com/NixOS/nixpkgs/commit/a5dcd41e27c1330eb8333951938511c3a5e97f9c.

I'm fine with including this as well, however please do so as separate commit.
Sorry for the extra work here, but a commit called claiming to reformat files must not have functional changes in it.

Comment (osnyx) [reply pkgs/servers/sql/percona-server/8_0.nix]: Now the diff mismatch results from explicitly _not_ backporting the changes from e0464e47880a69896f0fb1810f00e0de469f770a.

**Review by Ma27 — APPROVED** [review]
FTR: innovation was designed to be the latest feature-release, hence the upgrade to 8.4 is acceptable here (similar examples are linux_latest or nixVersions.git).

The diff in the cherry-pick job is unrelated (`stdenv.hostPlatform.is*` vs `stdenv.is*` mostly).
